### DISCUSSION TITLE
For LiveGrep in current buffer (LiveGrepB) there is no reason show file name in output

### DISCUSSION THREAD

**Discussion by damanis** [linrongbin16/fzfx.nvim#435] [General]
For LiveGrep in current buffer (LiveGrepB) there is no reason to show file name in output. So, it gives more screen space for found lines.

**Comment by linrongbin16** [comment]
hi @damanis, thanks for suggestion. I will take a look into it later.

**Reply by linrongbin16** [reply]
hi @damanis , it seems remove file name cannot be simply done by remove the `-H` flags, it also relate to previewer configs, and previewer label configs.

But indeed I am thinking, only provide the very generic `providers`, `previewers` things is stilling too much engineering effort when users are trying to custom their own searcher.

I'm thinking provide some APIs and help users more easier to config their own searching behavior.

**Comment by damanis** [comment]
@linrongbin16 ,
I think it's not search behavior, but shown search results only. You parse ripgrep output, then you can skip or add any part when build output window.
Of course, there are others implementation possible.

**Reply by linrongbin16** [reply]
parse ripgrep output and remove the `filename` part is actually .... not quite easy in current architecture, it will do too much hack and break the whole architecture.

for now the only modifications on command output lines are prepend the icons by filetypes.

but I will try to indeed improve the customization, it should a core value of this plugin.

**Comment by damanis** [comment]
@linrongbin16 
May be as a temporary solution the `-I` can be added to `rg` command for current buffer? But it prevent preview to work.

**Reply by linrongbin16** [reply]
hi @damanis, let me take a look later.

**Reply by linrongbin16** [reply]
hi @damanis , sorry for being so late, I submit #620 to remove filename in live grep's buffer mode.

Should be done in several days.

Let's continue this in #611 .

**Reply by linrongbin16** [reply]
hi @damanis , I had add `FzfxBufLiveGrep` command for a **No Filename** live grep on current buffer in #622 . Please pull main branch and try!
